### PR TITLE
Working NameCrawler into built-in Java hashmaps.

### PR DESCRIPTION
**Pull request by learner-long-life**

### CODE REVIEW

File: dsa-23au/java-dsa/datasets/src/test/java/dev/codewithfriends/NameCrawlerTest.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package dev.codewithfriends;
```
Comment (JonR2002) [inline dsa-23au/java-dsa/datasets/src/test/java/dev/codewithfriends/NameCrawlerTest.java:1]: Overall with the project, what is your main goal with the name database?

File: dsa-23au/java-dsa/datasets/src/test/java/dev/codewithfriends/NameCrawlerTest.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+{
+
+    @Test
+    public void testGetNamesByLetterPage()
+    {
+        NameCrawler.loadHeadersMap();
+        Map<String,String> meaningsMap = NameCrawler.getNamesByLetterPage('q', 1);
+        assertEquals(100, meaningsMap.size());
+        assertEquals("Night Blooming Flower", meaningsMap.get("Quynh"));
+    }
```
Comment (JonR2002) [inline dsa-23au/java-dsa/datasets/src/test/java/dev/codewithfriends/NameCrawlerTest.java:23]: This line is using an assertion to check if the size of the meaningsMap is equal to 100. It's verifying that the method has returned 100 names starting with 'q' on the first page.

File: dsa-23au/java-dsa/datasets/src/test/java/dev/codewithfriends/NameCrawlerTest.java (hunk 1 above)
Comment (learner-long-life) [reply dsa-23au/java-dsa/datasets/src/test/java/dev/codewithfriends/NameCrawlerTest.java:1]: The goal is to crawl a large, interesting dataset that we can use to profile, compare, and measure the performance of different data structures. Perhaps even have a name recommendation service: given some qualities you want to convey or imbue, what are some suggested names?

**Comment by learner-long-life** [comment]
Closed and merged here https://github.com/TheEvergreenStateCollege/upper-division-cs/commit/e1ee253b11c2ec2d4c2e43a00e3e726fd1c165e3

**Pull request closed without merging**